src/main.cpp

```diff
@@ -638,22 +638,145 @@ unsigned int LimitOrphanTxSize(unsigned int nMaxOrphans) EXCLUSIVE_LOCKS_REQUIRE
+    int nMinHeight = 0;
+    int64_t nMinTime = 0;
+    // Will remain equal to true if all inputs are finalized
+    // (CTxIn::SEQUENCE_FINAL).
+    bool fFinalized = true;
+
+    BOOST_FOREACH(const CTxIn& txin, tx.vin) {
+        // Set a flag if we witness an input that isn't finalized.
+        if (txin.nSequence == CTxIn::SEQUENCE_FINAL)
+            continue;
+        else
+            fFinalized = false;
+
+        // Do not enforce sequence numbers as a relative lock time
+        // unless we have been instructed to, and a view has been
+        // provided.
+        if (!(fEnforceBIP68 && pCoinsView))
+            continue;
+
+        // Sequence numbers with the most significant bit set are not
+        // treated as relative lock-times, nor are they given any
+        // consensus-enforced meaning at this point.
+        if (txin.nSequence & CTxIn::SEQUENCE_LOCKTIME_DISABLED_FLAG)
+            continue;
+
+        // Fetch the UTXO corresponding to this input.
+        //
+        // If the UTXO is not found, proceed to the next because we
+        // will not be able to do the relative lock-time calculation.
+        // This should never happen in consensus code.
+        //
+        // The better solution would be to fail, but that requires
+        // rewriting many pieces of non-consensus code that call
+        // IsFinalTx or CheckFinalTx to collect this information
+        // and pass it in somehow. This would not be simple to
+        // back-port for the soft-fork, however.
+        if (!pCoinsView->GetCoins(txin.prevout.hash, coins))
+            continue;
+
+        // coins.nHeight is MEMPOOL_HEIGHT (an absurdly high value)
+        // if the parent transaction was from the mempool. We can't
+        // know what height it will have once confirmed, but we
+        // assume it makes it in the same block.
+        int nCoinHeight = std::min(coins.nHeight, nBlockHeight);
+
+        if (txin.nSequence & CTxIn::SEQUENCE_LOCKTIME_SECONDS_FLAG) {
+            // In two locations that follow we make reference to
+            // chainActive.Tip(). To prevent a race condition, we
+            // store a reference to the current tip.
+            //
+            // Note that it is not guaranteed that indexBestBlock will
+            // be consistent with the passed in view. The proper thing
+            // to do is to have the view return time information about
+            // UTXOs.
+            const CBlockIndex& indexBestBlock = *chainActive.Tip();
+
+            // The only time the negative branch of this conditional
+            // is executed is when the prior output was taken from the
+            // mempool, in which case we assume it makes it into the
+            // same block (see above).
+            int64_t nCoinTime = (nCoinHeight <= (indexBestBlock.nHeight+1))
+                              ? indexBestBlock.GetAncestor(nCoinHeight-1)->GetMedianTimePast()
```

Comment:
Pasting @maaku 's response on the commit :

> No, GetMedianTimePast is preferred because it is not a free choice by the miner who included the transaction being spent. BIP 113 changes the other end of the interval for both nLockTime and nSequence to be GetMedianTimePast, which is a soft-fork change, but changing the start of the interval in that way would be a hard-fork change. So we start from the very beginning using GetMedianTimePast.

Ok it is clear thanks !